### ISSUE TITLE
Opening local file in File Editor on Windows 'downloads' file instead

### ISSUE THREAD

**Issue by lexnederbragt** [jupyter/help#140]
One can open local (on-disk) text-files from the Notebook Dashboard. If so, the file will be opened in the [File Editor](http://jupyter-notebook.readthedocs.io/en/latest/ui_components.html#file-editor). I was teaching this recently and the students running Windows (with Anaconda installed) had a different behaviour. Instead of opening the file they were asked what they wanted to do with it, and the only useful option led to the file being 'downloaded' while it was already on their harddisk. It is hard to find anything useful when googling this issue so I thought to post it here. Is this a known problem?

**Comment by minrk** [comment]
I suspect this was mime-type related. What was the type of the file?

On the affected machine, what was the result of:

```python
import mimetypes
mimetypes.guess_type(filename)
```
?

**Comment by takluyver** [comment]
In the latest version of notebook, it should be possible to work around this by selecting the file and using the 'edit' button at the top of the file list. But yeah, we depend on the system indicating a text mimetype to know to open files in the editor, and this can be affected by other things installed on the system (especially on Windows, I think).

**Comment by lexnederbragt** [comment]
@minrk Affected machine showed this:

`('application/vnd.ms-excel', None)`

It was a `.csv` file, which may be relevant

**Comment by minrk** [comment]
Yes, that's definitely relevant. It looks like Excel has registered a (wrong) mime-type for CSV files. The new behavior in notebook 5.0 gives you the opportunity to explicitly select `edit`, which will open the editor regardless of mime-type, so there's a bypass for the cases where the system reports a non-text mime-type for text files.
